src/backend/database_models/conversation.py

```diff
@@ -2,23 +2,41 @@
 from uuid import uuid4
 
 from sqlalchemy import ForeignKey, Index, PrimaryKeyConstraint, String, UniqueConstraint
+from sqlalchemy.dialects.postgresql import ARRAY
 from sqlalchemy.orm import Mapped, mapped_column, relationship
 
 from backend.database_models.base import Base
-from backend.database_models.file import File
 from backend.database_models.message import Message
 
 
+class ConversationFileAssociation(Base):
+    __tablename__ = "conversation_files"
+
+    conversation_id: Mapped[str] = mapped_column(
+        ForeignKey("conversations.id", ondelete="CASCADE")
+    )
+    user_id: Mapped[str] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"))
+    file_id: Mapped[str] = mapped_column(String, default=None, nullable=False)
+    conversation: Mapped["Conversation"] = relationship(
+        "Conversation", back_populates="conversation_file_associations"
+    )
+
+    __table_args__ = (
+        UniqueConstraint("conversation_id", "file_id", name="unique_conversation_file"),
+    )
+
+
 class Conversation(Base):
     __tablename__ = "conversations"
 
-    id: Mapped[str] = mapped_column(String, default=lambda: str(uuid4()))
+    id: Mapped[str] = mapped_column(String, default=lambda: str(uuid4()), unique=True)
```

Comment:
Oh I see, I'm not sure why there was a overriden `id` column existing. Can you delete this column and make it use the `Base` model's `id` field?

Reply:
i believe the override is due to this table having a composite PK (id+user_id)

Reply:
Ah right, ty